radicle-node/src/control.rs

```diff
@@ -17,12 +17,22 @@ use crate::service::FetchResult;
 pub enum Error {
     #[error("failed to bind control socket listener: {0}")]
     Bind(io::Error),
+    #[error("invalid socket path specified: {0}")]
```

Comment:
This is invalid because the directory doesn't exist, right? Maybe it would be more helpful to state that to help the end user remedy the situation.

`<tangent>` I wish there was some mechanism for writing copy for errors (even templates) that you could link to from the `error` macro. That way you could have really informative errors without cluttering the code itself. Dhall does iirc and I quite liked that approach `</tangent>`

Reply:
> Actually no, it's if the path doesn't have a parent, ie. I think it'll only happen if `/` is passed in :D

Ok, then maybe we should say that :smile: `"the socket path '{0}' must have a parent, i.e. it should not be '/'"`

> Yeah, tbh I'm still generally unsatisfied with rust errors; I still find myself getting the dreaded `File not found (os error 2)` sometimes with no other context and I'm like shrug

The wooooooorst

Reply:
> Ok, then maybe we should say that smile `"the socket path '{0}' must have a parent, i.e. it should not be '/'"`

I almost feel like returning an easter egg if this happens :thinking: